### PR TITLE
Add an option to listen for HTTPS

### PR DESCRIPTION
**Pull request by nono**
It will be useful for https://github.com/cozy/cozy-proxy/issues/217

### CODE REVIEW

File: main.coffee [hunk 1]
```diff
@@ -179,6 +179,23 @@ americano._loadPlugins = (options, app, callback) ->
         callback()
 
 
+# Listen for http (or https) connections
+americano._listen = (app, options, callback) ->
+    if options.ssl
+        https = require 'https'
+        crtPath = process.env.SSL_CRT_PATH or '/etc/cozy/server.crt'
```
Comment (aenario) [inline main.coffee]: Americano is supposed to not be tightly linked to cozy. May be allows CRT_PATH & KEY_PATH to be passed as options and move the `/etc/cozy` defaults to the cozy-proxy.

File: main.coffee [hunk 2]
```diff
@@ -179,6 +179,23 @@ americano._loadPlugins = (options, app, callback) ->
...
+        keyPath = process.env.SSL_KEY_PATH or '/etc/cozy/server.key'
+        tlsOptions =
+            cert: fs.readFileSync crtPath
+            key:  fs.readFileSync keyPath
```
Comment (aenario) [inline main.coffee]: They are other options which might prove useful for americano user, may be we can accept a options.tls.xxx object, extends it with the readFiles and use that instead

**Comment by nono** [comment]
@aenario is it better now?

**Comment by aenario** [comment]
Released & Published as v0.4.3

**Pull request merged**